blog/02-02-2023-connect-and-interact-with-celo-using-web3onboard-library/connect-and-interact-with-celo-using-web3onboard-library.md

```diff
@@ -0,0 +1,1106 @@
+
+```bash
+>...
+
+  async function handleConnect() {
+    if (isUserAuthenticated) return;
+    try {
+      await web3Onboard?.connectWallet().then(walletStates => {
+        if (walletStates.length) {
+          setAuthentication(true);
+        }
+      });
+    } catch (error) {
+      console.log("Connect Error", error);
+    }
+  }
+```
+
+![image](images/15.png)
+
+- Create a logout function that signs user out of the dapp when they explicitly call it. Their wallets are not completely logged out. The library provides us with an interface that does a few things like:
+  - Disconnect all user's wallets from the dapp.
+  - Connect another wallet.
+  - Display information about the dapp to users.
+  - Display Celo balances.
+  - Display wallet address.
+  - Copy address.
+  - Display current connected chain/network.
+  - Switch between available networks
+
+```bash
+>...
+
+  async function logout() {
+    if (!isUserAuthenticated) throw new Error("User not authenticated");
+    await disconnect().then(() => {
+      setAuthentication(false);
+    });
+  }
+```
+![image](images/14.png)
+
+Your `pages/index.tsx` should look like this.
+
+```bash
+import React from "react";
+import App from "../components/App";
+import LandingPage from "../components/LandingPage";
+import { useConnectWallet, useSetChain, useWallets } from "@web3-onboard/react";
+import { web3Onboard } from "@/components/apis/web3Onboard/setup";
+
+export default function Home() {
+  const [isUserAuthenticated, setAuthentication] = React.useState<boolean>(false);
+
+  const [, connect, , disconnect] = useConnectWallet();
+
+  async function handleConnect() {
+    if (isUserAuthenticated) return;
+    try {
+      await web3Onboard?.connectWallet().then(walletStates => {
+        if (walletStates.length) {
+          setAuthentication(true);
+        }
+      });
+    } catch (error) {
+      console.log("Connect Error", error);
+    }
+  }
+
+  async function logout() {
+    if (!isUserAuthenticated) throw new Error("User not authenticated");
+    await disconnect().then(() => {
+      setAuthentication(false);
+    });
+  }
+
+  return <>{!isUserAuthenticated ? <LandingPage isUserAuthenticated={isUserAuthenticated} handleConnect={handleConnect} /> : <App logout={logout} reconnect={connect} />}</>;
+}
+
+```
+
+## components/ConnectButton.tsx
+
+Make a new file name `ConnectButton.tsx` inside `components` folder. We are simply separating the 'Button' component as against what we currently have in `components/LandingPage.tsx`.
+
+Since we already initialized web3Onboard, we can use any of the `@web3-onboard/react` hook anywhere in the application. Since connecting users' wallet is done via an asynchronous function, the 'connecting' property returned by the 'useConnectWallet' hook helps us deactivate the connectButton when the connection has been invoked but not yet ready so users don't continiously click the button.
+
+```bash
```

Comment:
```js ?

Reply:
Isn't this supposed to ```js?

Reply:
Please read more on the guidelines for reviewing. This is the correct format.